packages/manager/apps/dedicated-servers/src/pages/listing/index.tsx

```diff
@@ -50,6 +54,12 @@ export default function Layout() {
     shell.ux.hidePreloader();
   }, []);
 
+  useEffect(() => {
+    const hasClusterFeature = features?.['dedicated-server:cluster'];
+    const hasClusterData = isSuccess && isSuccessCluster && data?.length > 0;
+    setHasCluster(hasClusterFeature && hasClusterData);
+  }, [isSuccessCluster, isSuccess, features, data]);
```

Comment:
As features and data are objects this could cause unnecessary re-renders so relaying on its properties
```suggestion
  }, features?.['dedicated-server:cluster'], data?.length]);
```

Reply:
this is why before I use just the status of call , success and you suggest to add the object